rust/hyperlane-core/src/types/announcement.rs

```diff
@@ -44,44 +41,17 @@ impl Announcement {
         )
     }
 
-    /// EIP-191 compliant hash of the signing hash of the checkpoint.
-    pub fn eth_signed_message_hash(&self) -> H256 {
-        hash_message(self.signing_hash())
-    }
-
-    /// Sign an checkpoint using the specified signer
-    pub async fn sign_with<S: Signer>(self, signer: &S) -> Result<SignedAnnouncement, S::Error> {
-        let signature = signer
-            .sign_message_without_eip_155(self.signing_hash())
-            .await?;
-        Ok(SignedAnnouncement {
-            announcement: self,
+    async fn sign_with(
```

Comment:
Yes because the size of `self` cannot be known in the trait implementation. So it would work if I boxed it but that just feels dumb. Hmm, oh, I might be able to move the signing functions to the signer instead of in the signable.